File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hm = {}

        for i in range(len(nums)):
            hm[nums[i]] = hm.get(nums[i], 0) + 1
            if hm[nums[i]] > 1:
                return True
        
        return False
        

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> hm = new HashMap<>();

        for (int num: nums) {
            hm.merge(num, 1, Integer::sum);
        }

        for (int val: hm.values()) {
            if(val > 1) return true;
        }

        return false;
    }
}